src/components/Button/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { Loading } from '@components/Loading'
+import { HTMLAttributes } from 'react'
+import styled, { css, DefaultTheme } from 'styled-components'
+
+export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  children: string
+  size: 'large' | 'medium'
+  disabled?: boolean
+  isLoading?: boolean
+  block: boolean
+}
+
+const sizeModifiers = {
+  medium: (theme: DefaultTheme) => css`
+    height: ${theme.spacings.large};
+    font-size: ${theme.fonts.sizes.large};
+  `,
+  large: (theme: DefaultTheme) => css`
+    height: ${theme.spacings.xlarge};
+    font-size: ${theme.fonts.sizes.xlarge};
+  `,
+}
+
+const Wrapper = styled.button<Omit<ButtonProps, 'children'>>`
+  ${({ theme, size, disabled, block }) => css`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: ${block ? '100%' : 'auto'};
+    padding: 0 1rem;
```

Comment:
Não recomendo utilizar rem para espaçamento. REM é o tipo de unidade de medida que você utiliza quando quer que as coisas escalem conforme as configurações de acessibilidade do navegador do usuário e isso normalmente serve para coisas como font, letter spacing, line height, etc. 
Esse artigo explica bem os meus pontos sobre isso:
https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility/